### PR TITLE
:pencil2: [REFACTOR] : Added section for Don't use functional widget.

### PR DESCRIPTION
**Pull request by DevarshRanpara**
:pencil2: [REFACTOR] : Added section for Don't use functional widget.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,3 +86,57 @@ MaterialApp(
...
+```
+And in this case Widget tree looks like this
+```dart
+ClassWidget
+  Container
+    ClassWidget
+      Container
+```
+As we can see here, if we use Widgets, framework understands it in better way and UI becames easy to inspect.
+For mode info follow [this](https://stackoverflow.com/a/53234826/11445644) answer from stackoverflow
```
Comment (bhavik-multipl) [inline README.md:142]: should be `more`,
Can you make this change else LGTM.

**Review by bhavik-multipl — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -85,3 +86,57 @@ MaterialApp(
 You can do much more with the Theme, which is defining your custom TextField, Card, Bottom Navigation Bar themes for once and use it straight away in the app. Checkout an example of using Theme over [here](https://github.com/ibhavikmakwana/supabase_playground/blob/master/lib/values/theme.dart)
 
 >In Addition, In a large app/project you might end up having a multipl Themes not just light and dark mode themes but Custom themes for some specific widgets/components as well so you can always create a separate class/file and maintain all the themes over there.
+
+### dont-use-functional-widgets
+
+We usually have a situation where we need to saperate out UI code from the widget, But we avoid creating a saperate widget and use function which returns Widget.
+This practice have some benefits, like you don't need to pass all parameters in your new widget, You have less code and less files. But this approch may cause issue when you want to inspect your widget. Let's see this in depth.
```
Comment (ibhavikmakwana) [inline README.md]: should be `separate` instead of `saperate`

Comment (ibhavikmakwana) [reply README.md]: approach instead of `approch`

File: README.md (hunk 1 above)
Comment (ibhavikmakwana) [inline README.md]: mode should be `more` and `becomes` instead of `becames`

**Review by ibhavikmakwana — CHANGES_REQUESTED** [review]

**Comment by ibhavikmakwana** [comment]
just some typo fix needed else LGTM

**Comment by DevarshRanpara** [comment]
@ibhavikmakwana I have fixed typos and updated PR.

**Review by ibhavikmakwana — APPROVED** [review]

**Pull request merged**